core/src/utils.jl

```diff
@@ -12,25 +12,97 @@ function valid_edge_types(db::DB)::Bool
     return !errors
 end
 
-"Return a directed graph, and a mapping from source and target nodes to edge fid."
-function create_graph(
-    db::DB,
-    edge_type_::String,
-)::Tuple{DiGraph, Dictionary{Tuple{Int, Int}, Int}, Dictionary{Int, Tuple{Symbol, Symbol}}}
+"""
+Return a directed metagraph with data of nodes (NodeMetadata):
+- Node type (snake case)
+- Allocation network ID
+
+and data of edges (EdgeMetadata):
+- Edge ID (EdgeID)
+- type (flow/control)
+"""
+function create_graph(db::DB)::MetaGraph
     node_rows = execute(db, "select fid, type from Node")
-    nodes = dictionary((fid => Symbol(type) for (; fid, type) in node_rows))
-    graph = DiGraph(length(nodes))
     edge_rows = execute(db, "select fid, from_node_id, to_node_id, edge_type from Edge")
-    edge_ids = Dictionary{Tuple{Int, Int}, Int}()
-    edge_connection_types = Dictionary{Int, Tuple{Symbol, Symbol}}()
-    for (; fid, from_node_id, to_node_id, edge_type) in edge_rows
-        if edge_type == edge_type_
-            add_edge!(graph, from_node_id, to_node_id)
-            insert!(edge_ids, (from_node_id, to_node_id), fid)
-            insert!(edge_connection_types, fid, (nodes[from_node_id], nodes[to_node_id]))
+    graph = MetaGraph(
+        DiGraph();
+        label_type = NodeID,
+        vertex_data_type = NodeMetadata,
+        edge_data_type = EdgeMetadata,
+        graph_data = nothing, # In theory all remaining connectivity data could be put here
```

Comment:
If you think this would be an improvement, it'd be good to make a follow up issue/task.